### PR TITLE
 Visitor can view articles by category

### PR DESCRIPTION
**Pull request by psnoya**
[Visitor can view articles by category CLIENT](https://www.pivotaltracker.com/story/show/176053440)
### User Story
```
As a visitor
In order to find an article I want to read
I would like to be able to view articles by category
```
## Changes proposed in this pull request:
* Front-end category component -created with Menu items
* Add index by category function in getArticles module
## What I have learned working on this feature:
* How to work with redux in the menu bar

### CODE REVIEW

File: cypress/integration/userCanViewArticlesByCategory.feature.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+describe("A user can,", () => {
+  describe("successfully,", () => {
```
Comment (ViktorHek) [inline cypress/integration/userCanViewArticlesByCategory.feature.js]: you can delete describe("successfully") if you don't have a sad path

File: src/components/CategoryMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
```
Comment (ViktorHek) [inline src/components/CategoryMenu.jsx]: remove useState

File: src/modules/getArticles.js [hunk 3]
```diff
@@ -2,18 +2,23 @@ import axios from "axios";
 
 const getArticles = {
   async index(dispatch) {
-    let result = await axios.get("/articles");
-    dispatch({type: "SET_NEWS_FEED", payload: result.data.articles})
+    const result = await axios.get("/articles");
```
Comment (Ehugo2000) [inline src/modules/getArticles.js]: Do you need to use a try-catch block here and in the category request?

File: src/modules/getArticles.js [hunk 4]
```diff
@@ -2,18 +2,23 @@ import axios from "axios";
...
     try {
-      let response = await axios.get(`/articles/${articleId}`);
+      const response = await axios.get(`/articles/${articleId}`);
       dispatch({ type: "VIEW_ARTICLE", payload: response.data.article });
     } catch (error) {
       dispatch({ type: "ERROR_MESSAGE", payload: error.response.data.message });
     }
   },
+
+  async index_by_category(categoryId, dispatch) {
```
Comment (emtalen) [inline src/modules/getArticles.js:25]: are you really finding the category by the id in the api?

**Review by emtalen — CHANGES_REQUESTED** [review]

File: src/modules/getArticles.js (hunk 4 above)
Comment (KyunginNa) [reply src/modules/getArticles.js:25]: We were using label, so we fixed it in the api!

File: src/App.jsx [hunk 5]
```diff
@@ -2,18 +2,20 @@ import React from "react";
...
 
 const App = () => {
   return (
     <>
-    <Header/>
+      <Header />
+      <CategoryMenu />
       <Switch>
         <Route exact path="/" component={DisplayArticlesList}></Route>
         <Route exact path="/articles/:id" component={DisplayArticle}></Route>
```
Comment (Ehugo2000) [inline src/App.jsx:16]: Why did you remove the category Route?

File: src/components/DisplayArticlesList.jsx [hunk 6]
```diff
@@ -6,7 +6,7 @@ import { useDispatch, useSelector } from "react-redux";
 
 const DisplayArticlesList = () => {
   const dispatch = useDispatch();
-  const { newsFeed } = useSelector(state => state);
+  const { newsFeed } = useSelector((state) => state);
```
Comment (Ehugo2000) [inline src/components/DisplayArticlesList.jsx]: Might be able to clean up code, remove () around ```(state)```.

**Review by Ehugo2000 — COMMENTED** [review]
Left some comments

File: src/modules/getArticles.js (hunk 3 above)
Comment (KyunginNa) [reply src/modules/getArticles.js]: It is because we didn't create an error handler in the api

File: src/App.jsx (hunk 5 above)
Comment (KyunginNa) [reply src/App.jsx:16]: It didn't work, so we're still trying to figure it out. Yet, it doesn't need to be in this user story so we deleted it for now.

File: src/components/DisplayArticlesList.jsx (hunk 6 above)
Comment (KyunginNa) [reply src/components/DisplayArticlesList.jsx]: Removed it :)

File: src/modules/getArticles.js (hunk 3 above)
Comment (emtalen) [reply src/modules/getArticles.js]: But I agree with Erik here, you should have a catch block always, you could get a 500 error for example

**Review by emtalen — CHANGES_REQUESTED** [review]

File: src/modules/getArticles.js (hunk 3 above)
Comment (KyunginNa) [reply src/modules/getArticles.js]: Oh okay! We added try and catch blocks!

**Review by emtalen — APPROVED** [review]
